### ISSUE TITLE
Decimals

### ISSUE THREAD

**Issue by remster** [julianpeeters/avrohugger#29]
Decimals are quite solid in the avro spec (http://avro.apache.org/docs/1.7.7/spec.html#Decimal) and I need a BigDecimal at some point of the processing - ideally in the generated case class. I understand that this isn't supported at present. 

If indeed it isn't, is it anywhere near the roadmap? If not, how difficult/easy do you think it will be to retrofit? Is that something you would be happy to accept, with some hand holding, as a contribution?

**Comment by julianpeeters** [comment]
Decimals are not yet on the roadmap, but I agree that they should be, and would be happy to help guide you if you'd like to contribute.

I suspect it will be pretty easy, similar in effort to how the BYTES datatype was added:
code + unit tests (`sbt test`) - https://github.com/julianpeeters/avrohugger/commit/616915e697fae3257ee997f342a1815292679ad3
integration tests (`sbt scripted`) - https://github.com/julianpeeters/avrohugger/commit/d024dbbf5d09c34b0e253173b42d99eebb383c6f

I might start work by navigating to one of the scripted integration test projects (SpecificSerializationTests) and setting up a test that used a .java version of the simple specific record class, then converting that .java class to .scala by hand, just to see what I was aiming for. I might then use that .scala example as a unit test in avrohugger-core that I'd try to make pass. Finally, I might go back and pull out the dummy .scala example class we added to the SpecificSerializationTests integration test, and test that it is instead generated automatically  from the root project with `sbt-scripted` (make sure that you've bumped the project to a SNAPSHOT or something, or else `scripted` will pick up the current release from maven central rather than your modified version it publishes to local ivy).

Hope this helps, let me know if you run into any questions.

**Comment by julianpeeters** [comment]
Oh, and the projects in in the SpecificSerialiationTests integrations tests depend on the [sbt-avrohugger plugin](https://github.com/julianpeeters/sbt-avrohugger), so to see any of the changes that you make in avrohugger reflected in the integration test projects, you'll have to have a clone of the plugin, upgraded to your current avrohugger SNAPSHOT, `publish-local` under a new version (the dependency version is "latest.integration" so the latest version is picked up automatically).

This setup is admittedly a rigamarole, but it lets the integration tests live in the project that causes them to pass or not. Other ideas are certainly welcome.

**Comment by remster** [comment]
Thanks! I'll give it a go.
Forked. Saw and fixed
https://github.com/julianpeeters/avrohugger/compare/master...remster:fix-tests
the test failing with the package name problem that you may or may not be
aware of.
BTW, you seem to be ignoring all errors:

Try {
   tool.run(
     System.in, System.out, System.err, Arrays.asList(args:
_*).subList(1, args.length));
 }

and thus, a botched input produces no output or a complaint. Is this an
intention or the backlog's bottom?
In other words, would you mind if I exposed these problems - I am trying to
use your tool in a build chain.
It will help to know if it succeeded.

**Comment by julianpeeters** [comment]
Thanks for noticing and fixing that package name error.

There's no good reason the errors are ignored, I'm all for exposing them.

**Comment by remster** [comment]
I am sorry Julian, but i won't be pursuing this until we can migrate our
emr to Scala 2.11.
And because we need to run our spark on 2.10, we became victim to the dreaded
22 c'tor arguments limit https://issues.scala-lang.org/browse/SI-7296 for
2.10 - we're having a table with 94 columns.
We need to revert to java nd traditional avro-tools.

I may be coming back.
regards and sorry for the fuss.

**Comment by julianpeeters** [comment]
No worries at all, you have already been enormously helpful.

**Comment by mariussoutier** [comment]
@remster Avrohugger supports Scala 2.10 and 22 fields since 0.12.1.

**Comment by mariussoutier** [comment]
Re: decimals, right now I'm using longs to circumvent any decimal encoding issues and divide by 100.0 (when dealing with currencies for example) when processing the data later in the application.

**Comment by llyjy21** [comment]
actually, the avro-tools supports bigdecimal, why the plugin dosen't yet ?

**Comment by julianpeeters** [comment]
@llyjy21 avro-tools uses the Apache Avro compiler to generate Java, which supports BigDecimal. Avrohugger has its own compiler in order to generate Scala code, but support for logical types has not yet been implemented.

**Comment by eirirlar** [comment]
+1 for logical types.

**Comment by julianpeeters** [comment]
Shoutout to @raulraja for contributing decimal support, now available in avrohugger version 1.0.0-RC9 and sbt-avrohugger version 2.0.0-RC9 :clap:
